r/src/metadata.cpp

```diff
@@ -0,0 +1,215 @@
+#include <Rcpp.h>
+#include "rrrow_types.h"
+
+// [[Rcpp::plugins(cpp11)]]
+
+using namespace Rcpp ;
+
+template <typename... String >
+xptr_DataType metadata( const std::shared_ptr<arrow::DataType>& ptr, String... strings ){
+  xptr_DataType res( new std::shared_ptr<arrow::DataType>(ptr) ) ;
+  res.attr("class") = CharacterVector::create( ptr->name(), strings... ) ;
+  return res ;
+}
+
+xptr_DataType metadata_integer( const std::shared_ptr<arrow::DataType>& ptr ){
+  return metadata( ptr, "arrow::Integer", "arrow::Number", "arrow::PrimitiveCType", "arrow::FixedWidthType", "arrow::DataType" ) ;
```

Comment:
Should we use common clang-format style files for both the C++ codebase and the Rcpp files?

Reply:
Not sure what you mean

Reply:
See https://github.com/apache/arrow/blob/master/cpp/.clang-format. These files do not conform to the C++ code formatting used in the rest of the project